django/core/management/commands/shell.py

```diff
@@ -79,6 +62,27 @@ def python(self, options):
                 except Exception:
                     traceback.print_exc()
 
+        # By default, this will set up readline to do tab completion and to read and
+        # write history to the .python_history file, but this can be overridden by
+        # $PYTHONSTARTUP or ~/.pythonrc.py.
+        try:
+            sys.__interactivehook__()
+        except Exception:
+            # Match the behavior of the cpython shell where an error in
+            # sys.__interactivehook__ prints a warning and the exception and continues.
+            print("Failed calling sys.__interactivehook__")
+            traceback.print_exc()
+
+        # Set up tab completion for objects imported by $PYTHONSTARTUP or
+        # ~/.pythonrc.py.
+        try:
+            import readline
+            import rlcompleter
+            readline.set_completer(rlcompleter.Completer(imported_objects).complete)
+        except ImportError:
+            pass
```

Comment:
From @inglesp's [comment](https://github.com/django/django/pull/13911/files#r565608154) above:
> We're not calling `site.enablerlcompleter`, but instead we're calling whatever the value of `sys.__interactivehook__` is

My [original mention of the `site` module](https://github.com/django/django/pull/13911/files#r565417290) was ambiguous. What I meant by 
> lean directly on the `site` module

was something like this

```suggestion
        # ~/.pythonrc.py. We intentionally ignore sys.flags.isolated so that
        # readline is available from the command line in otherwise isolated
        # environments for which sys.__interactivehook__ might not already
        # have been set by the site module.
        try:
            import site
        except ImportError:
            pass
        else:
            old_hook = getattr(sys, "__interactivehook__", None)
            try:
                site.enablerlcompleter()
                sys.__interactivehook__()
            finally:
                if old_hook is not None:
                    sys.__interactivehook__ = old_hook
```
Again, I'm not entirely convinced calling `site.enablerlcompleter` itself is advisable. But the point is that at least `site.enablerlcompleter` includes all the libedit compatibility shims.